CMakeLists.txt

```diff
@@ -32,11 +32,17 @@ add_compile_options($<$<CONFIG:Debug>:-Wall>)
 add_compile_options($<$<CONFIG:Debug>:-Wno-parentheses>)
 add_compile_options($<$<CONFIG:Debug>:-Wno-uninitialized>)
 add_compile_options($<$<CONFIG:Debug>:-Wmissing-prototypes>)
-if (NOT APPLE)
+if(NOT APPLE)
     add_compile_options($<$<CONFIG:Debug>:-Wsystem-headers>)
 endif()
 add_compile_options($<$<CONFIG:Release>:-Wuninitialized>)
 add_compile_options($<$<CONFIG:Release>:-Wno-dangling-else>)
+if(USE_ADDRESS_SANITIZER)
```

Comment:
Please make it an `option` and split it into a different PR. The executable hangs on my machine BTW, needs some investivation.

Reply:
Another option is to let me add an analyzer configuration type later (`CMAKE_CONFIGURATION_TYPES`) based on http://www.stablecoder.ca/2018/02/01/analyzer-build-types.html

Reply:
Understood. I will remove the commit modifying `CMakeLists.txt` for now. Since I do not currently have the time or resources to test AddressSanitizer builds properly, I will defer to someone else to add the build file support. My apologies about this!